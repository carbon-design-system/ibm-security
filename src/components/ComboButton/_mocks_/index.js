/**
 * @file Combo button mocks.
 * @copyright IBM Security 2019
 */

import { ArrowRight20, FolderDetails20 } from '@carbon/icons-react';

const actions = [
  {
    label: 'Start a task',
    iconDescription: 'arrow right',
    renderIcon: ArrowRight20,
    onClick: () => {},
  },
  {
    label: 'Add to folder',
    renderIcon: FolderDetails20,
    onClick: () => {},
  },
];

export default actions;
