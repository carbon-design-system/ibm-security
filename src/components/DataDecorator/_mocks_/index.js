/**
 * @file Data decorator mocks.
 * @copyright IBM Security 2019
 */

const props = {
  type: 'IP',
  value: '10.0.0.0',
  score: 0,
  href: '#',
};
const midLine = {
  type: 'MD5',
  value: '0f3deda483df5e5f8043ea20297d243b',
  score: 30,
  href: '#',
};

export default props;

export { props, midLine };
