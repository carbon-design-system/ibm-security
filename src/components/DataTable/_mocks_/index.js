/* eslint-disable react/prop-types */
/**
 * @file Data table mocks.
 * @copyright IBM Security 2019
 */

import React from 'react';
import {
  Table,
  TableBody,
  TableCell,
  TableContainer,
  TableHead,
  TableHeader,
  TableRow,
  TableSelectAll,
  TableSelectRow,
} from '../../..';

const headers = [
  {
    key: 'lastSighted',
    header: 'Last sighted',
  },
  {
    key: 'name',
    header: 'Name',
  },
  {
    key: 'type',
    header: 'Type',
  },
  {
    key: 'attackPhase',
    header: 'Attack phase',
  },
  {
    key: 'method',
    header: 'Method',
  },
  {
    key: 'missing',
    header: 'Missing',
  },
];

const isSelectable = false;
const isSortable = false;

const rows = [
  {
    id: '7c34f4507156b35',
    lastSighted: '11/28/2018',
    name: 'IBM X-Force',
    type: 'IPv6',
    attackPhase: 'Initial Compromise',
    method: 'Phishing',
  },
  {
    id: '38347816a51b0c5a8',
    lastSighted: '06/28/2017',
    name: 'AppX',
    type: 'Hash',
    attackPhase: 'Establish Foothold',
    method: 'Deploy backdoor',
  },
  {
    id: '1d6116a2d1d03b4',
    lastSighted: '05/30/2018',
    name: 'Bigfix',
    type: 'URL',
    attackPhase: 'Escalate Privileges',
    method: 'Credential dumping',
  },
  {
    id: 'c52b0419981221765',
    lastSighted: '12/12/2018',
    name: 'Am I Affected',
    type: 'Hash',
    attackPhase: 'Cross-Site Scripting',
    method: 'Malicious Code',
  },
  {
    id: '697626e6f703131d',
    lastSighted: '11/11/2018',
    name: 'QRadar',
    type: 'Hash',
    attackPhase: 'SQL Injection Attack',
    method: 'SQL Vulnerabilities',
  },
  {
    id: 'd568f12fa7b7cac1',
    lastSighted: '12/09/2018',
    name: 'iris',
    type: 'Hash',
    attackPhase: 'Volume Based Attacks',
    method: 'ICMP floods',
  },
  {
    id: '697626e6f70313554',
    lastSighted: '12/10/2018',
    name: 'TISP',
    type: 'Hash',
    attackPhase: 'Application Layer Attacks',
    method: 'GET/POST floods',
  },
  {
    id: '1dd568f12fa7b7cac2',
    lastSighted: '12/11/2018',
    name: 'CLX',
    type: 'Hash',
    attackPhase: 'Protocol Attacks',
    method: 'SYN Floods',
  },
  {
    id: '697626e6f703131',
    lastSighted: '10/12/2018',
    name: 'X-Force Plus',
    type: 'Hash',
    attackPhase: 'Malicious Links',
    method: 'Social Engineering',
  },
  {
    id: 'dd568f12fa7b7cac0',
    lastSighted: '12/16/2018',
    name: 'BigFix & iris',
    type: 'Hash',
    attackPhase: 'Exploit Kits',
    method: 'Scripts',
  },
  {
    id: '697626e6f70313',
    lastSighted: '01/12/2018',
    name: 'Cases',
    type: 'Hash',
    attackPhase: 'Malicious Email',
    method: 'Social Engineering',
  },
];

const page = 1;
const pageSize = 5;
const pageSizes = [5, 10, 20];

/**
 * Render method.
 * @returns {string} The string to render.
 */
const render = ({
  rows,
  headers,
  getHeaderProps,
  getRowProps,
  getTableProps,
  getSelectionProps,
}) => (
  <TableContainer>
    <Table {...getTableProps()}>
      {headers && (
        <TableHead>
          <TableRow>
            <TableSelectAll {...getSelectionProps()} />
            {headers.map(header => (
              <TableHeader
                key={header.id}
                {...getHeaderProps({ header, isSortable })}
              >
                {header.header}
              </TableHeader>
            ))}
          </TableRow>
        </TableHead>
      )}
      <TableBody>
        {rows.map(row => (
          <TableRow {...getRowProps({ row })} key={row.id}>
            <TableSelectRow {...getSelectionProps({ row })} />
            {row.cells.map(cell => (
              <TableCell key={cell.id}>
                {cell.value ? cell.value : 'â€“'}
              </TableCell>
            ))}
          </TableRow>
        ))}
      </TableBody>
    </Table>
  </TableContainer>
);

export {
  headers,
  isSelectable,
  isSortable,
  page,
  pageSize,
  pageSizes,
  render,
  rows,
};
