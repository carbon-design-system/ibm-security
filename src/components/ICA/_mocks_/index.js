/**
 * @file Important content area mocks.
 * @copyright IBM Security 2019
 */

const value = 10;

export default {
  label: 'Label',
  value,
  total: value * value,
};
