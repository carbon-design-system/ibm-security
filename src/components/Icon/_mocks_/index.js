/**
 * @file Icon mocks.
 * @copyright IBM Security 2019 - 2021
 */

import { Add16 } from '@carbon/icons-react';

const path = 'M0';
const renderIcon = Add16;

export { path, renderIcon };
