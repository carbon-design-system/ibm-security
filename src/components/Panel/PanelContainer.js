/**
 * @file Panel container.
 * @copyright IBM Security 2019
 */

/* eslint-disable no-useless-computed-key, react/require-default-props */

import Close20 from '@carbon/icons-react/lib/close/20';

import classnames from 'classnames';
import PropTypes, { func } from 'prop-types';
import React, { Component, createRef, Fragment } from 'react';
import { createPortal } from 'react-dom';

import requiredIfGivenPropExists from 'carbon-components-react/lib/prop-types/requiredIfGivenPropExists';
import setupGetInstanceId from 'carbon-components-react/lib/tools/setupGetInstanceId';

import { getComponentNamespace } from '../../globals/namespace';
import * as defaultLabels from '../../globals/nls';

import { isClient, isNode } from '../../globals/utils/capabilities';
import { focusFirstElement, trapTabFocus } from '../../globals/utils/focus';

import Button from '../Button';
import IconButton from '../IconButton';

export const namespace = getComponentNamespace('panel');
const getInstanceId = setupGetInstanceId();

/**
 * Panel container component.
 * @param {Object.<string, *>} props Panel container props.
 * @returns {PanelContainer} Panel container instance.
 */
export default class PanelContainer extends Component {
  constructor(props) {
    super(props);
    if (isClient()) {
      this.container = this.getContainer();
      this.containerClass = `${namespace}__container`;
      this.element = document.createElement('section');

      this.element.classList.add(namespace);

      this.previousFocus = document.activeElement;
      this.rootNode = this.props.rootNode;

      const { className } = this.props;

      if (className) {
        this.element.classList.add(className);
      }
    }
  }

  state = { bodyMargin: 0 };

  componentDidMount() {
    if (isClient()) {
      if (!this.container) {
        this.container = document.createElement('div');
        this.rootNode.appendChild(this.container);
      }

      this.container.appendChild(this.element);
      if (this.container.childElementCount === 1) {
        this.rootNode.classList.toggle(this.containerClass);
      }

      focusFirstElement(this.element);
      this.element.addEventListener('keydown', this.handleKeyPress);

      this.setBodyMargin();
    }
  }

  componentWillUnmount() {
    if (isClient()) {
      this.container.removeChild(this.element);

      if (this.container.childElementCount === 0) {
        this.rootNode.classList.toggle(this.containerClass);
        this.rootNode.removeChild(this.container);
      }

      this.element.removeEventListener('keydown', this.handleKeyPress);
      this.previousFocus.focus();
    }
  }

  /**
   * Retrieves the element to render the panel container in to via a portal.
   * @returns {HTMLElement} The element to render the panel container in to via a portal.
   */
  getContainer = () => {
    if (isClient()) {
      const panels = document.getElementsByClassName(namespace);

      if (panels.length > 0) {
        return panels[0].parentNode;
      }
    }

    return null;
  };

  /**
   * Sets the body margin to match the height of the header for fixed scrolling.
   */
  setBodyMargin() {
    const { current: footer } = this.footer;

    this.setState({
      bodyMargin: {
        top: this.header.current.clientHeight,
        bottom: footer && footer.clientHeight,
      },
    });
  }

  panelInstanceId = `panel-${getInstanceId()}`;
  panelTitleId = `${namespace}__title--${this.panelInstanceId}`;
  panelSubtitleId = `${namespace}__subtitle--${this.panelInstanceId}`;

  footer = createRef();
  header = createRef();

  /**
   * Handles the key press focus trap.
   * @param {SyntheticEvent} event The event fired when a key has been pressed while the panel container is in focus.
   */
  handleKeyPress = event => {
    if (event.key === 'Tab') {
      trapTabFocus(this.element, event);
    } else if (!this.props.disableEscape && event.key === 'Escape') {
      this.props.closeButton.onClick();
    }
  };

  renderPanelContainer = ({
    labels: {
      PANEL_CONTAINER_PRIMARY_BUTTON,
      PANEL_CONTAINER_SECONDARY_BUTTON,
      PANEL_CONTAINER_CLOSE_BUTTON,
    },
  }) => {
    const {
      children,
      closeButton,
      primaryButton,
      renderFooter,
      secondaryButton,
      subtitle,
      title,
      hasScrollingContent,
    } = this.props;

    const hasFooter = renderFooter || primaryButton;

    const ariaLabel = this.props['aria-label'] || title || subtitle;

    const getAriaLabelledBy = title ? this.panelTitleId : this.panelSubtitleId;

    const hasScrollingContentProps = hasScrollingContent
      ? {
          tabIndex: 0,
          role: 'region',
          'aria-label': ariaLabel,
          'aria-labelledby': getAriaLabelledBy,
        }
      : {};

    return (
      <div role="dialog" aria-label={ariaLabel} aria-modal="true">
        <header ref={this.header} className={`${namespace}__header`}>
          {title && (
            <div className={`${namespace}__header__container--title`}>
              <div
                id={this.panelTitleId}
                className={`${namespace}__header--title`}
              >
                {title}
              </div>
              {subtitle && (
                <div
                  id={this.panelSubtitleId}
                  className={`${namespace}__header--subtitle`}
                >
                  {subtitle}
                </div>
              )}
            </div>
          )}
          <IconButton
            id={closeButton.id}
            className={`${namespace}__button--close`}
            label={PANEL_CONTAINER_CLOSE_BUTTON}
            onClick={closeButton.onClick}
            renderIcon={closeButton.icon || Close20}
            tooltip={false}
          />
        </header>
        <section
          className={classnames(`${namespace}__body`, {
            [`${namespace}__body--footer`]: hasFooter,
          })}
          style={{
            marginTop: `${this.state.bodyMargin.top}px`,
            marginBottom: `${this.state.bodyMargin.bottom}px`,
          }}
          {...hasScrollingContentProps}
          aria-labelledby={getAriaLabelledBy}
        >
          {children}
        </section>
        {hasFooter && (
          <footer ref={this.footer} className={`${namespace}__footer`}>
            {renderFooter ? (
              renderFooter()
            ) : (
              <Fragment>
                {secondaryButton && (
                  <Button
                    id={secondaryButton.id}
                    className={`${namespace}__footer__button ${namespace}__footer__button--secondary`}
                    disabled={secondaryButton.isDisabled}
                    iconDescription={secondaryButton.iconDescription}
                    kind="secondary"
                    onClick={secondaryButton.onClick}
                    renderIcon={secondaryButton.icon}
                  >
                    {PANEL_CONTAINER_SECONDARY_BUTTON}
                  </Button>
                )}
                <Button
                  id={primaryButton.id}
                  className={`${namespace}__footer__button`}
                  disabled={primaryButton.isDisabled}
                  iconDescription={primaryButton.iconDescription}
                  onClick={primaryButton.onClick}
                  renderIcon={primaryButton.icon}
                >
                  {PANEL_CONTAINER_PRIMARY_BUTTON}
                </Button>
              </Fragment>
            )}
          </footer>
        )}
      </div>
    );
  };

  render() {
    const { closeButton, primaryButton, secondaryButton, labels } = this.props;

    const componentLabels = {
      ...defaultLabels.labels,
      ...labels,
      ...defaultLabels.filterFalsey({
        PANEL_CONTAINER_PRIMARY_BUTTON:
          (primaryButton && primaryButton.label) || '',
        PANEL_CONTAINER_SECONDARY_BUTTON:
          (secondaryButton && secondaryButton.label) || '',
        PANEL_CONTAINER_CLOSE_BUTTON: (closeButton && closeButton.label) || '',
      }),
    };
    return (
      isClient() &&
      createPortal(
        this.renderPanelContainer({ labels: componentLabels }),
        this.element
      )
    );
  }
}

const buttonType = PropTypes.shape({
  id: PropTypes.string,
  onClick: PropTypes.func,
  label: PropTypes.string,
  isDisabled: PropTypes.bool,
  icon: PropTypes.object,
  iconDescription: PropTypes.string,
});

PanelContainer.propTypes = {
  /** @type {ReactNode} The children of the panel container. */
  children: PropTypes.node,

  /** @type {string} Class name. */
  className: PropTypes.string,

  /** @type {Object<Object>} An object list of close button props. */
  closeButton: buttonType.isRequired,

  /** @type {boolean} Set to true to disable the 'Escape' key from closing the panel. */
  disableEscape: PropTypes.bool,

  /** @type {Object<Object>} An object list of primary button props. */
  primaryButton: buttonType,

  /** @type {Object<Object>} An object list of secondary button props. */
  secondaryButton: buttonType,

  /** @type {string} Child elements. */
  subtitle: PropTypes.node,

  /** @type {string} Child elements. */
  title: PropTypes.node,

  /** @type {function} Footer render prop. */
  renderFooter: func,

  /** @type {string} Root node to attach the panel to. */
  rootNode: isNode() ? PropTypes.instanceOf(Node) : PropTypes.any,

  labels: defaultLabels.propType,

  /**
   * Specify whether the panel contains scrolling content
   */
  hasScrollingContent: PropTypes.bool,

  /**
   * Required props for the accessibility label of the header
   */
  ['aria-label']: requiredIfGivenPropExists(
    'hasScrollingContent',
    PropTypes.string
  ),
};

PanelContainer.defaultProps = {
  children: null,
  className: null,
  disableEscape: false,
  primaryButton: null,
  secondaryButton: undefined,
  subtitle: undefined,
  title: undefined,
  renderFooter: null,
  rootNode: isNode() ? document.body : undefined,
  labels: {},
  hasScrollingContent: false,
};

/* eslint-enable */
