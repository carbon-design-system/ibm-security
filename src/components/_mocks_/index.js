/**
 * @file Global mocks.
 * @copyright IBM Security 2018
 */

import React from 'react';

import random from '../../globals/random';

const label = 'Label';

const lorem =
  'Lorem ipsum dolor sit, amet consectetur adipisicing elit. Eveniet rerum aliquid perferendis, nulla nam ad excepturi, iure earum dolorum ipsa quae cum voluptatibus nemo quis debitis, aperiam repudiandae dolore deleniti.';

const icon =
  'data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZpZXdCb3g9IjAgMCAzMiAzMiI+ICA8cGF0aCBmaWxsPSIjZmZmIiBkPSJNMTYgMThINmEyIDIgMCAwIDEtMi0yVjZhMiAyIDAgMCAxIDItMmgxMGEyIDIgMCAwIDEgMiAydjEwYTIgMiAwIDAgMS0yIDJ6TTYgNnYxMGgxMFY2em0yMCA2djRoLTR2LTRoNG0wLTJoLTRhMiAyIDAgMCAwLTIgMnY0YTIgMiAwIDAgMCAyIDJoNGEyIDIgMCAwIDAgMi0ydi00YTIgMiAwIDAgMC0yLTJ6bTAgMTJ2NGgtNHYtNGg0bTAtMmgtNGEyIDIgMCAwIDAtMiAydjRhMiAyIDAgMCAwIDIgMmg0YTIgMiAwIDAgMCAyLTJ2LTRhMiAyIDAgMCAwLTItMnptLTEwIDJ2NGgtNHYtNGg0bTAtMmgtNGEyIDIgMCAwIDAtMiAydjRhMiAyIDAgMCAwIDIgMmg0YTIgMiAwIDAgMCAyLTJ2LTRhMiAyIDAgMCAwLTItMnoiLz48L3N2Zz4=';

const id = '0';

const disabled = true;

const invalid = true;

const invalidText = 'Invalid text';

/**
 * Returns a child component.
 */
const children = <span>{label}</span>;

const className = 'class';

/**
 * Creates a URL for mock data.
 * @param {string} [hash=''] - A hash to pass to the URL method.
 * @returns {string} The mock data URL.
 */
const url = (hash = '') => `#${hash}`;

/**
 * Creates a `href` for mock data.
 * @returns {string} The `href` to return.
 */
const href = () => url();

export {
  children,
  className,
  disabled,
  href,
  icon,
  id,
  invalid,
  invalidText,
  label,
  lorem,
  random,
  url,
};
