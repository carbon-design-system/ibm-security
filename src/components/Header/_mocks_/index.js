/**
 * @file Header mocks.
 * @copyright IBM Security 2018
 */

import { url } from '../../_mocks_';
import labels from '../locales/en/Header.json';

const hasSearch = true;
const hasSearchError = false;

const links = {
  product: url(),
  profile: url(),
  registration: url(),
  sign_in: url(),
  sign_out: url(),
  edit_profile: url(),
  notifications_preferences: url(),
  notifications_view_all: url(),
};

const accounts = new Array(3).fill().map((name = `Company`, id) => ({
  id: id.toString(),
  name: `${name} ${id + 1}`,
}));

const notifications = [
  {
    id: `notification__0`,
    description: 'A new request requires approval.',
    href: url(),
    product: 'Application 3',
    datetime: 'Fri Dec 21 2018 18:57:08 GMT-0600 (Central Standard Time)',
    label: '11min ago',
  },
  {
    id: `notification__1`,
    description: 'There have been 17 failed sign-ins in the last 3 days.',
    href: url(),
    product: 'Application 2',
    datetime: 'Wed May 01 2019 22:22:52 GMT-0500 (Central Daylight Time)',
    label: '1hr ago',
  },
  {
    id: `notification__2`,
    description: 'There are 3 unread messages in the queue.',
    href: url(),
    product: 'Application 3',
    datetime: 'Fri Jul 06 2018 00:17:49 GMT-0500 (Central Daylight Time)',
    label: '3hrs ago',
  },
  {
    id: `notification__3`,
    description: 'Review the results of the latest scan.',
    href: url(),
    product: 'Application 3',
    datetime: 'Sun Apr 14 2019 20:32:17 GMT-0500 (Central Daylight Time)',
    label: '3hrs ago',
  },
  {
    id: `notification__4`,
    description: 'A new administrator has been added to your account.',
    href: url(),
    product: 'Application 1',
    datetime: 'Sat Oct 20 2018 19:07:04 GMT-0500 (Central Daylight Time)',
    label: '4hrs ago',
  },
  {
    id: `notification__5`,
    description: 'There are 5 items that require your review.',
    href: url(),
    product: 'Application 2',
    datetime: 'Tue Dec 25 2018 18:14:28 GMT-0600 (Central Standard Time)',
    label: '5hrs ago',
  },
  {
    id: `notification__7`,
    description:
      'There are 103 threats found in the last scan. Please click to see the details.',
    href: url(),
    product: 'Application 2',
    datetime: 'Tue Nov 25 2018 18:14:28 GMT-0600 (Central Standard Time)',
    label: '10hrs ago',
  },
  {
    id: `notification__8`,
    description:
      'There are 3 ongoing attacks. Please click to take actions immediately.',
    href: url(),
    product: 'Application 2',
    datetime: 'Tue Oct 25 2018 18:14:28 GMT-0600 (Central Standard Time)',
    label: '24hrs ago',
  },
];

const profile = {
  image_url: null,
  name: {
    first_name: 'Sample',
    surname: 'User',
  },
};

const profileWithAccount = Object.assign({}, profile, {
  email: 'example@ibm.com',
  account: accounts[0],
});

const profileWithAccountLongName = Object.assign({}, profile, {
  email: 'example@ibm.com',
  account: {
    ...accounts[0],
    name: `${accounts[0].name} plus more text to make the name super super long`,
  },
});
const defaultSearchValue = '127.0.0.1';

export {
  defaultSearchValue,
  hasSearch,
  hasSearchError,
  labels,
  links,
  accounts,
  notifications,
  profile,
  profileWithAccount,
  profileWithAccountLongName,
};
