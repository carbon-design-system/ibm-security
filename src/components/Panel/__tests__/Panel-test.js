/**
 * @file Panel tests.
 * @copyright IBM Security 2019, 2021
 */

import { render, fireEvent, screen } from '@testing-library/react';
import React, { Fragment } from 'react';

import { label } from '../_mocks_';

import {
  Button,
  Panel,
  PanelContainer,
  PanelContent,
  PanelController,
} from '../../..';

import { namespace } from '../PanelContainer';

describe('Panel', () => {
  let panelActive;

  const { fn } = jest;

  const onClose = fn(() => {
    if (panelActive) {
      panelActive = false;
    }
  });

  const onOpen = fn(() => {
    if (!panelActive) {
      panelActive = true;
    }
  });

  const panelRef = React.createRef();
  const panelElement = (props) => (
    <Panel
      ref={panelRef}
      onClose={onClose}
      onOpen={onOpen}
      shouldPanelOpen={!panelActive}
      render={({ active, toggleActive }) => (
        <Fragment>
          <Button onClick={toggleActive}>toggle panel</Button>
          <PanelController active={active}>
            <PanelContainer
              title={label}
              subtitle={label}
              closeButton={{ label, onClick: toggleActive }}>
              <PanelContent>Panel content</PanelContent>
            </PanelContainer>
          </PanelController>
        </Fragment>
      )}
      {...props}
    />
  );

  beforeEach(() => {
    panelActive = false;
  });

  describe('Rendering', () => {
    it('renders correctly', () => {
      const { asFragment } = render(panelElement());
      expect(screen.queryByText('Panel content')).toBeNull();
      expect(asFragment()).toMatchSnapshot();
    });

    it("renders the HTML of the node's subtree", () => {
      render(panelElement());
      expect(panelRef.current.render()).toMatchSnapshot();
    });

    it('renders the open panel', async () => {
      const { asFragment } = render(panelElement());
      await fireEvent.click(
        screen.getByRole('button', { name: /toggle panel/i })
      );
      expect(screen.getByText('Panel content')).toBeInTheDocument();
      expect(asFragment()).toMatchSnapshot();
    });

    it('does not render the open panel if `shouldPanelOpen` is false', async () => {
      const { asFragment } = render(panelElement({ shouldPanelOpen: false }));
      await fireEvent.click(
        screen.getByRole('button', { name: /toggle panel/i })
      );
      expect(screen.queryByText('Panel content')).toBeNull();
      expect(asFragment()).toMatchSnapshot();
    });

    it('renders the correct body margin', async () => {
      render(panelElement());
      await fireEvent.click(
        screen.getByRole('button', { name: /toggle panel/i })
      );
      const body = screen.getByRole('region');
      expect(body).toHaveStyle(`margin-top: 0px`);
    });
  });

  describe('Events', () => {
    it('should invoke `onOpen`', async () => {
      render(panelElement());
      await fireEvent.click(
        screen.getByRole('button', { name: /toggle panel/i })
      );
      expect(onOpen).toBeCalled();
    });

    it('should invoke `onClose`', async () => {
      render(panelElement());
      await fireEvent.click(
        screen.getByRole('button', { name: /toggle panel/i })
      );
      await fireEvent.click(
        screen.getByRole('button', { name: /toggle panel/i })
      );
      expect(onClose).toBeCalled();
    });
  });
});
