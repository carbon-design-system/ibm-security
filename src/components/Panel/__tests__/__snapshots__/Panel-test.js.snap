// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel Rendering does not render the open panel if \`shouldPanelOpen\` is false 1`] = `
<Fragment>
  <Button
    dangerDescription="danger"
    disabled={false}
    kind="primary"
    largeText={null}
    loading={false}
    onClick={[Function]}
    size="default"
    tabIndex={0}
    tooltipAlignment="center"
    tooltipPosition="top"
    type="button"
  >
    Label
  </Button>
  <PanelController
    active={false}
  >
    <PanelContainer
      className={null}
      closeButton={
        Object {
          "label": "Label",
          "onClick": [Function],
        }
      }
      disableEscape={false}
      hasScrollingContent={false}
      labels={Object {}}
      primaryButton={null}
      renderFooter={null}
      rootNode={<body />}
      subtitle="Label"
      title="Label"
    >
      <PanelContent>
        Label
      </PanelContent>
    </PanelContainer>
  </PanelController>
</Fragment>
`;

exports[`Panel Rendering renders correctly 1`] = `
<Fragment>
  <Button
    dangerDescription="danger"
    disabled={false}
    kind="primary"
    largeText={null}
    loading={false}
    onClick={[Function]}
    size="default"
    tabIndex={0}
    tooltipAlignment="center"
    tooltipPosition="top"
    type="button"
  >
    Label
  </Button>
  <PanelController
    active={false}
  >
    <PanelContainer
      className={null}
      closeButton={
        Object {
          "label": "Label",
          "onClick": [Function],
        }
      }
      disableEscape={false}
      hasScrollingContent={false}
      labels={Object {}}
      primaryButton={null}
      renderFooter={null}
      rootNode={<body />}
      subtitle="Label"
      title="Label"
    >
      <PanelContent>
        Label
      </PanelContent>
    </PanelContainer>
  </PanelController>
</Fragment>
`;

exports[`Panel Rendering renders the HTML of the node's subtree 1`] = `
<button
  class="bx--btn bx--btn--primary"
  tabindex="0"
  type="button"
>
  Label
</button>
`;

exports[`Panel Rendering renders the open panel 1`] = `
<Fragment>
  <Button
    dangerDescription="danger"
    disabled={false}
    kind="primary"
    largeText={null}
    loading={false}
    onClick={[Function]}
    size="default"
    tabIndex={0}
    tooltipAlignment="center"
    tooltipPosition="top"
    type="button"
  >
    Label
  </Button>
  <PanelController
    active={true}
  >
    <PanelContainer
      className={null}
      closeButton={
        Object {
          "label": "Label",
          "onClick": [Function],
        }
      }
      disableEscape={false}
      hasScrollingContent={false}
      labels={Object {}}
      primaryButton={null}
      renderFooter={null}
      rootNode={<body />}
      subtitle="Label"
      title="Label"
    >
      <PanelContent>
        Label
      </PanelContent>
    </PanelContainer>
  </PanelController>
</Fragment>
`;

exports[`Panel Rendering renders the primary button variation 1`] = `
<Fragment>
  <Button
    dangerDescription="danger"
    disabled={false}
    kind="primary"
    largeText={null}
    loading={false}
    onClick={[Function]}
    size="default"
    tabIndex={0}
    tooltipAlignment="center"
    tooltipPosition="top"
    type="button"
  >
    Label
  </Button>
  <PanelController
    active={true}
  >
    <PanelContainer
      className={null}
      closeButton={
        Object {
          "label": "Label",
          "onClick": [Function],
        }
      }
      disableEscape={false}
      hasScrollingContent={false}
      labels={Object {}}
      primaryButton={null}
      renderFooter={null}
      rootNode={<body />}
      subtitle="Label"
      title="Label"
    >
      <PanelContent>
        Label
      </PanelContent>
    </PanelContainer>
  </PanelController>
</Fragment>
`;

exports[`Panel Rendering renders the secondary button variation 1`] = `
<Fragment>
  <Button
    dangerDescription="danger"
    disabled={false}
    kind="primary"
    largeText={null}
    loading={false}
    onClick={[Function]}
    size="default"
    tabIndex={0}
    tooltipAlignment="center"
    tooltipPosition="top"
    type="button"
  >
    Label
  </Button>
  <PanelController
    active={true}
  >
    <PanelContainer
      className={null}
      closeButton={
        Object {
          "label": "Label",
          "onClick": [Function],
        }
      }
      disableEscape={false}
      hasScrollingContent={false}
      labels={Object {}}
      primaryButton={null}
      renderFooter={null}
      rootNode={<body />}
      subtitle="Label"
      title="Label"
    >
      <PanelContent>
        Label
      </PanelContent>
    </PanelContainer>
  </PanelController>
</Fragment>
`;

exports[`Panel Rendering renders the secondary button variation 2`] = `
<Fragment>
  <Button
    dangerDescription="danger"
    disabled={false}
    kind="primary"
    largeText={null}
    loading={false}
    onClick={[Function]}
    size="default"
    tabIndex={0}
    tooltipAlignment="center"
    tooltipPosition="top"
    type="button"
  >
    Label
  </Button>
  <PanelController
    active={true}
  >
    <PanelContainer
      className={null}
      closeButton={
        Object {
          "label": "Label",
          "onClick": [Function],
        }
      }
      disableEscape={false}
      hasScrollingContent={false}
      labels={Object {}}
      primaryButton={null}
      renderFooter={null}
      rootNode={<body />}
      subtitle="Label"
      title="Label"
    >
      <PanelContent>
        Label
      </PanelContent>
    </PanelContainer>
  </PanelController>
</Fragment>
`;
